main/util/src/mill/util/Jvm.scala

```diff
@@ -538,8 +538,8 @@ object Jvm {
           Result.Failure(
             s"Failed to load ${if (sources) "source " else ""}dependencies" + errorDetails
           )
-        case Right(res) =>
-          Result.Success(res)
+        case Right(artifacts) =>
+          Result.Success(ArtifactResolution(resolution, artifacts))
```

Comment:
Maybe `coursier.Fetch.Result` could be used for that, like
```suggestion
          Result.Success(Fetch.Result(resolution, artifacts.fullDetailedArtifacts0, artifacts.fullExtraArtifacts))
```

Reply:
And we could rename `CoursierModule#artifacts` to `CoursierModule#fetch` then

Reply:
Used Fetch.Result and renamed.

Reply:
@alexarchambault Ping. Updated to Fetch.Result